### PR TITLE
Bug/thrust limit and voltage failsafe

### PR DESCRIPTION
**Pull request by PizzaAllTheWay**
Have limited thrusters and checked that it physically limited thrust, it works!
Have implemented a failsafe so that if voltage measurements dip bellow or above a certain threshold the motors stop momentarily until voltage values are in the green zone again and everything works fine. Tested, works very well :)
Have added thrust limiting and voltage failsafe as a beluga.yaml file variable that can be conveniently changed and modified

### CODE REVIEW

File: thruster_interface/scripts/thruster_interface_node.py [hunk 1]
```diff
@@ -175,10 +190,10 @@ def validate_and_limit_thrust(self, thrust_msg):
...
 
     def limit_pwm(self, pwm):
-        if pwm > 1900:
-            return 1900
-        if pwm < 1100:
-            return 1100
+        if pwm > self.pwm_limit_max:
+            return self.pwm_limit_max
+        if pwm < self.pwm_limit_min:
+            return self.pwm_limit_min
```
Comment (theBadMusician) [inline thruster_interface/scripts/thruster_interface_node.py]: I'd like the thruster limiting to be re-implemented as a percentage-based value instead.

File: thruster_interface/scripts/thruster_interface_node.py [hunk 2]
```diff
@@ -264,8 +301,12 @@ def thrust_cb(self, thrust_msg):
...
     )
     NUM_THRUSTERS = rospy.get_param("/propulsion/thrusters/num", default=8)
-    THRUST_OFFSET = rospy.get_param("/propulsion/thrusters/offset", default=[0, 0, 0, 0, 0, 0, 0, 0])
-    THRUSTER_DIRECTION = rospy.get_param("/propulsion/thrusters/direction", default=[1, 1, 1, 1, 1, 1, 1, 1])
+    THRUST_OFFSET = rospy.get_param(
+        "/propulsion/thrusters/offset", default=[0, 0, 0, 0, 0, 0, 0, 0]
+    )
+    THRUSTER_DIRECTION = rospy.get_param(
+        "/propulsion/thrusters/direction", default=[1, 1, 1, 1, 1, 1, 1, 1]
+    )
```
Comment (theBadMusician) [inline thruster_interface/scripts/thruster_interface_node.py]: Remove the default param values here - if offset values and directions cannot be loaded in from a config file, the thruster are not going to function as predicted i.e., won't spin in correct directions etc.

**Review by theBadMusician — CHANGES_REQUESTED** [review]
1. Don't use default rosparam values for functionality critical parameters.
2. Re-implement thruster limiting to percentage-based values, e.g., -70% to +70% instead of 1220 to 1780. Take into account thruster offset values.

File: thruster_interface/scripts/thruster_interface_node.py (hunk 2 above)
Comment (PizzaAllTheWay) [reply thruster_interface/scripts/thruster_interface_node.py]: Done and tested, it works as expected :)

**Pull request merged**